webcam-capture-drivers/driver-ipcam/src/main/java/com/github/sarxos/webcam/ds/ipcam/IpCamDevice.java

```diff
@@ -235,30 +86,71 @@ public IpCamDevice(String name, URL url, IpCamMode mode, IpCamAuth auth) {
 		}
 	}
 
-	public IpCamHttpClient getClient() {
-		return client;
-	}
-
-	protected static final URL toURL(String url) {
-
-		String base = null;
-		if (url.startsWith("http://")) {
-			base = url;
-		} else {
-			base = String.format("http://%s", url);
-		}
-
-		try {
-			return new URL(base);
-		} catch (MalformedURLException e) {
-			throw new WebcamException(String.format("Incorrect URL '%s'", url), e);
-		}
-	}
-
-	@Override
-	public String getName() {
-		return name;
-	}
+    protected static final URL toURL(String url) {
+
+        String base = null;
+        if (url.startsWith("http://")) {
+            base = url;
+        } else {
+            base = String.format("http://%s", url);
+        }
+
+        try {
+            return new URL(base);
+        } catch (MalformedURLException e) {
+            throw new WebcamException(String.format("Incorrect URL '%s'", url), e);
+        }
+    }
+
+    private void initThreadPool() {
+        queue = new ArrayBlockingQueue<Runnable>(1);
+        executorService =
+                new ThreadPoolExecutor(
+                        1,
+                        1,
+                        1,
+                        TimeUnit.SECONDS,
+                        queue,
+                        new ThreadFactory() {
+                            @Override
+                            public Thread newThread(Runnable r) {
+                                final String threadName = String.format("%s-reader", getName());
+                                Thread thread = new Thread(r, threadName);
+                                thread.setDaemon(true);
+                                return thread;
+                            }
+                        });
+        executorService.shutdown();

```

Comment:
Hi @bocse, I'm wondering why you are doing ExecutorService::shutdown() immediately after it's being initialized? I think this should be rather done in IpCamDevice::close().